extensions/amp-drilldown/0.1/amp-drilldown.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-drilldown-0.1.css';
+import {Keys} from '../../../src/utils/key-codes';
+import {Layout} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {closestAncestorElementBySelector} from '../../../src/dom';
+import {dev, userAssert} from '../../../src/log';
+import {isExperimentOn} from '../../../src/experiments';
+
+const TAG = 'amp-drilldown';
+
+// TODO(stevenye): add unit tests for click handlers and history.
+// https://github.com/ampproject/amphtml/issues/24668
+export class AmpDrilldown extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private @const {!Document} */
+    this.document_ = this.win.document;
+
+    /** @private @const {!Element} */
+    this.documentElement_ = this.document_.documentElement;
+
+    /** @private {Array} */
+    this.historyIds_ = [];
+
+    /** @private {?Element} */
+    this.currentSubmenu_ = null;
+
+    /** @private {function(!Event)} */
+    this.submenuOpenHandler_ = this.handleSubmenuOpenClick_.bind(this);
+
+    /** @private {function(!Event)} */
+    this.submenuCloseHandler_ = this.handleSubmenuCloseClick_.bind(this);
+  }
+
+  /** @override */
+  buildCallback() {
+    userAssert(isExperimentOn(this.win, 'amp-sidebar-v2'), 'Experiment is off');
```

Comment:
Nit: Since this error message is for clients of our library (and the performance team recently did some work to allow us to make verbose logs), make this something more descriptive like "Turning on the amp-sidebar-v2 experiment is necessary to use the amp-drilldown component."